### PR TITLE
Add example for strategy design pattern

### PR DESCRIPTION
**Pull request by amanjain97**
- Strategy pattern for chosing different algorithms for search engine. 
- this pattern can be used when we perform A/B test for different algorithms.

### CODE REVIEW

**Comment by anuragdhingra** [comment]
@amanjain97 
Since, you wrote it lets keep it as a second example for strategy pattern. 
But can you rebase it over my `feature/strategy-pattern` and move this file to the `strategy` folder?

**Comment by amanjain97** [comment]
@anuragdhingra Please check again.

File: strategy/example2.go [hunk 1]
```diff
@@ -0,0 +1,147 @@
+package main
+
+import (
+	"bufio"
+	"fmt"
+	"os"
+)
+
+// =========================== ANALYZER =================================
```
Comment (anuragdhingra) [inline strategy/example2.go]: Let's maybe follow a common commentary style guide, https://golang.org/doc/effective_go.html#commentary ?

File: strategy/example2.go [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+		fmt.Printf("Query is: %s", query)
+		fmt.Printf("Analyzed: %s", v4.analyzer.Analyze(query))
+		fmt.Printf("Search results: %s", v4.searcher.Search(query))
+		fmt.Printf("Query Metadata: %s", v4.metadataFetcher.Metadata(query))
+	} else {
+		fmt.Println("Please specify correct version")
+	}
+}
+
+// Youtube link: https://www.youtube.com/watch?v=v9ejT8FO-7I
```
Comment (anuragdhingra) [inline strategy/example2.go]: This is already available in `strategy.md` file, with the defintion for the pattern, lets skip it from here then.

**Review by anuragdhingra — DISMISSED** [review]
LGTM, except the above two comments.

File: strategy/example2.go (hunk 1 above)
Comment (amanjain97) [reply strategy/example2.go]: This is not package/ function comment. It was just meant to separate the components for visibility. 
Do you suggest to remove it ?

Comment (anuragdhingra) [reply strategy/example2.go]: Maybe having this with additional line comments over your interface or methods will be helpful.

Comment (amanjain97) [reply strategy/example2.go]: ok.

**Review by anuragdhingra — APPROVED** [review]

**Pull request merged**